stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/content/FileUploadContent.kt

```diff
@@ -87,39 +85,41 @@ public fun FileUploadItem(attachment: Attachment) {
                     color = ChatTheme.colors.textHighEmphasis
                 )
 
-                val uploadId = requireNotNull(attachment.uploadId)
-
-                val tracker = ProgressTrackerFactory.getOrCreate(uploadId)
-
-                val uploadProgress by tracker.currentProgress().collectAsState()
-                val isComplete by tracker.isComplete().collectAsState()
-                val maxValue = attachment.upload?.length() ?: tracker.maxValue
-
-                if (!isComplete) {
-                    Row(Modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically) {
-                        LoadingView(modifier = Modifier.size(12.dp))
-
-                        Spacer(modifier = Modifier.size(8.dp))
-
-                        val uploadedSize = (uploadProgress / 100F * maxValue).toLong()
+                when (val uploadState = attachment.uploadState) {
+                    is Idle -> {
+                        ProgressInfo(uploadedBytes = 0L, totalBytes = attachment.upload?.length() ?: 0L)
+                    }
+                    is InProgress -> {
+                        ProgressInfo(uploadedBytes = uploadState.bytesRead, totalBytes = uploadState.totalBytes)
+                    }
+                    else -> {
                         Text(
-                            text = stringResource(
-                                id = R.string.stream_compose_upload_progress,
-                                MediaStringUtil.convertFileSizeByteCount(uploadedSize),
-                                MediaStringUtil.convertFileSizeByteCount(maxValue),
-                            ),
+                            text = MediaStringUtil.convertFileSizeByteCount(attachment.upload?.length() ?: 0L),
                             style = ChatTheme.typography.footnote,
                             color = ChatTheme.colors.textLowEmphasis
                         )
                     }
-                } else {
-                    Text(
-                        text = MediaStringUtil.convertFileSizeByteCount(attachment.upload?.length() ?: 0L),
-                        style = ChatTheme.typography.footnote,
-                        color = ChatTheme.colors.textLowEmphasis
-                    )
                 }
             }
         }
     }
 }
+
+@Composable
```

Comment:
I'd just add KDocs here and maybe a named argument for `modifier` for the `Row`.

Otherwise, looks good!

Reply:
Added!